browser/ui/tabs/split_view_browser_data.cc

```diff
@@ -107,4 +108,72 @@ void SplitViewBrowserData::RemoveObserver(
   observers_.RemoveObserver(observer);
 }
 
+SplitViewBrowserData::OnTabDragEndedClosure::OnTabDragEndedClosure() = default;
+
+SplitViewBrowserData::OnTabDragEndedClosure::OnTabDragEndedClosure(
+    base::WeakPtr<SplitViewBrowserData> data,
+    base::OnceClosure closure)
+    : data_(data),
+      closure_(
+          std::make_unique<base::ScopedClosureRunner>(std::move(closure))) {
+  CHECK(data_);
+}
+
+SplitViewBrowserData::OnTabDragEndedClosure::OnTabDragEndedClosure(
+    SplitViewBrowserData::OnTabDragEndedClosure&& other) noexcept {
+  RunCurrentClosureIfNeededAndReplaceWith(std::move(other));
+}
+
+SplitViewBrowserData::OnTabDragEndedClosure&
+SplitViewBrowserData::OnTabDragEndedClosure::operator=(
+    SplitViewBrowserData::OnTabDragEndedClosure&& other) noexcept {
+  RunCurrentClosureIfNeededAndReplaceWith(std::move(other));
+  return *this;
+}
+
+void SplitViewBrowserData::OnTabDragEndedClosure::RunAndReset() {
+  if (closure_) {
+    this->closure_->RunAndReset();
+  }
+  data_.reset();
+}
+
+void SplitViewBrowserData::OnTabDragEndedClosure::
+    RunCurrentClosureIfNeededAndReplaceWith(OnTabDragEndedClosure&& other) {
+  if (this->data_.get() == other.data_.get()) {
+    // In case |this| and |other| are pointing at the same |data_|, just discard
+    // the old one. This means we've got the callback from the same Browser
+    // instance.
+    if (this->closure_) {
+      this->closure_->Release().Reset();
+    }
+  } else {
+    // Target Browser was changed, so we need to run the callback for the old
+    // target browser.
+    if (this->closure_) {
+      this->closure_->RunAndReset();
+    }
+  }
+
```

Comment:
how about adding `this->data_.reset();` here?
Then `other` will have empty `closure_` and `data_` after swap.
I think `other` should be empty after moving.